coordinator/src/services/translation/response/release/release-response.ts

```diff
@@ -60,24 +61,34 @@ function createPrescriptionsBundleParameter(
   }
 }
 
-export function translateReleaseResponse(releaseResponse: hl7V3.PrescriptionReleaseResponse): fhir.Parameters {
+export function translateReleaseResponse(
+  releaseResponse: hl7V3.PrescriptionReleaseResponse,
+  logger: pino.BaseLogger): fhir.Parameters {
   const releaseRequestId = releaseResponse.inFulfillmentOf.priorDownloadRequestRef.id._attributes.root
   const parentPrescriptions = toArray(releaseResponse.component)
     .filter(component => component.templateId._attributes.extension === SUPPORTED_MESSAGE_TYPE)
     .map(component => ({
       prescription: component.ParentPrescription,
       errors: verifySignature(component.ParentPrescription)
     }))
-  const passedPrescriptions = parentPrescriptions.filter(prescriptionResult => prescriptionResult.errors.length === 0)
+  const passedPrescriptionResources = parentPrescriptions
```

Comment:
Hi @MatthewPopat-NHS looks like we iterate over parentPrescriptions twice when we are filtering passed and failed prescriptions. If we used a parentPrescriptions.reduce() we could return an object which has a failedList and a passedList and only iterate over once.

I'm happy to talk about this on a call to explain further